packages/backend-common/src/database/connection.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import knex from 'knex';
+import { Config } from '@backstage/config';
+import { buildPgConnectionConfig } from './postgres';
+import { buildSqlite3ConnectionConfig } from './sqlite3';
+
+/**
+ * Creates a knex database connection
+ *
+ * @param config The database config
+ * @param dbName The optional database to connect to
+ */
+export function createDatabase(config: Config, dbName?: string) {
```

Comment:
how does this relate to possibly segment by schema? https://github.com/spotify/backstage/issues/1598#issuecomment-658278931